sdk/azurestackhci/Azure.ResourceManager.Hci/src/Generated/OfferData.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.Hci.Models;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.Hci
+{
+    /// <summary> A class representing the Offer data model. </summary>
+    public partial class OfferData : ResourceData
+    {
+        /// <summary> Initializes a new instance of OfferData. </summary>
+        public OfferData()
+        {
+            SkuMappings = new ChangeTrackingList<SkuMappings>();
+        }
+
+        /// <summary> Initializes a new instance of OfferData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="provisioningState"> Provisioning State. </param>
+        /// <param name="publisherId"> Identifier of the Publisher for the offer. </param>
+        /// <param name="content"> JSON serialized catalog content of the offer. </param>
+        /// <param name="contentVersion"> The API version of the catalog service used to serve the catalog content. </param>
+        /// <param name="skuMappings"> Array of SKU mappings. </param>
+        internal OfferData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, string provisioningState, string publisherId, string content, string contentVersion, IList<SkuMappings> skuMappings) : base(id, name, resourceType, systemData)
+        {
+            ProvisioningState = provisioningState;
+            PublisherId = publisherId;
+            Content = content;
+            ContentVersion = contentVersion;
+            SkuMappings = skuMappings;
+        }
+
+        /// <summary> Provisioning State. </summary>
+        public string ProvisioningState { get; }
+        /// <summary> Identifier of the Publisher for the offer. </summary>
+        public string PublisherId { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

Reply:
These are inherited as string from swagger. No change required.